torch/utils/data/datapipes/_typing.py

```diff
@@ -369,6 +395,15 @@ def wrap_generator(*args, **kwargs):
                         response = gen.send(request)
             except StopIteration as e:
                 return e.value
+            except DataPipeException:
+                raise
+            except Exception as e:
+                datapipe = args[0]
+                # Short version, but requires custom __repr__ for each DataPipe class
+                # msg = f"thrown by {datapipe}"
+                # Long version with all input arguments
+                msg = f"thrown by __iter__ of {datapipe.__class__.__name__}({get_input_args(datapipe)})"
```

Comment:
Thoughts on listing out all the input arguments? I think it makes it obvious to the users which DataPipe is responsible at the cost of having a longer message.